### PR TITLE
*: Update the row structure in executor

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: store/tikv/mock-tikv/executor.go [hunk 1]
```diff
@@ -343,28 +343,29 @@ func getRowData(columns []*tipb.ColumnInfo, colIDs map[int64]int, handle int64,
 	return values, nil
 }
 
-func extractColIDsInExpr(expr *tipb.Expr, columns []*tipb.ColumnInfo, collector map[int64]int) error {
+func extractColIDsInExpr(expr *tipb.Expr, columns []*tipb.ColumnInfo, collector []int64) ([]int64, error) {
```
Comment (coocood) [inline store/tikv/mock-tikv/executor.go]: The collector should be allocated with the length of the columns beforehand, doesn't need to be returned.

File: distsql/xeval/eval.go [hunk 2]
```diff
@@ -51,24 +51,35 @@ const (
...
 // Evaluator evaluates tipb.Expr.
 type Evaluator struct {
-	Row         map[int64]types.Datum // column values.
-	ColumnInfos []*tipb.ColumnInfo
+	Row map[int64]types.Datum // TODO: Remove this field after refactor cop_handler.
 
-	fieldTps   []*types.FieldType
-	valueLists map[*tipb.Expr]*decodedValueList
-	sc         *variable.StatementContext
+	RowNew      []types.Datum // column values
```
Comment (coocood) [inline distsql/xeval/eval.go]: We can name it `ColVals` directly.

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
cc @AndreMouche @hhkbp2

**Pull request merged**